libexec/pyenv

```diff
@@ -61,24 +61,6 @@ else
 fi
 export PYENV_ROOT
 
-# Transfer PYENV_FILE_ARG (from shims) into PYENV_DIR.
-if [ -z "${PYENV_DIR}" ]; then
-  if [ -n "${PYENV_FILE_ARG}" ]; then
-    if [ -L "${PYENV_FILE_ARG}" ]; then
-      PYENV_DIR="$(abs_dirname "${PYENV_FILE_ARG}")"
-    else
-      PYENV_DIR="${PYENV_FILE_ARG%/*}"
-    fi
-    export PYENV_DIR
-    unset PYENV_FILE_ARG
-  fi
-else
-  [[ $PYENV_DIR == /* ]] || PYENV_DIR="$PWD/$PYENV_DIR"
-  cd "$PYENV_DIR" 2>/dev/null || abort "cannot change working directory to \`$PYENV_DIR'"
-  PYENV_DIR="$PWD"
-  cd "$OLDPWD"
-fi
-
 if [ -z "${PYENV_DIR}" ]; then
```

Comment:
This is kinda off-topic, however, duplicating rbenv's behavior on handling of `RBENV_DIR` here might make sense at extent. At least it seems it's verifying the availability of given `RBENV_DIR` before exporting. It might be safer 🤔 

https://github.com/rbenv/rbenv/blob/d604acb78aeba583be95f08d45eeae430372beb9/libexec/rbenv#L60-L68

Reply:
All the `PYENV_DIR` handling code that I delete here is duplicated by the code just below that is missing in Rbenv.

What the deleted code does boils down to `PYENV_DIR = $(realpath PYENV_DIR)`.

Since that code below is sketchy, I'm fine with restoring the Rbenv lookalike and sorting the duplication out in a differrent PR.

Reply:
> Since that code below is sketchy, I'm fine with restoring the Rbenv lookalike and sorting the duplication out in a differrent PR.

Whoops, can't do that!

Rbenv's lookalike code was removed in 19e2b9588a3824e007a7689f9a918336768e120f to fix a breakage!

Reply:
Since this feature is broken by design, it should be removed. 
I have looked at the proposed code for a long time, and  have no questions.